pkg/cloud_provider/file/file.go

```diff
@@ -646,6 +658,99 @@ func (manager *gcfsServiceManager) StartResizeShareOp(ctx context.Context, obj *
 	return nil, nil
 }
 
-func (manager *gcfsServiceManager) WaitForOpWithOpts(ctx context.Context, opts PollOpts) error {
+func (manager *gcfsServiceManager) WaitForOpWithOpts(ctx context.Context, op string, opts PollOpts) error {
+	return nil
+}
+
+func (manager *gcfsServiceManager) GetOp(ctx context.Context, op string) (*filev1beta1.Operation, error) {
+	opInfo, err := manager.operationsService.Get(op).Context(ctx).Do()
+	if err != nil {
+		return nil, err
+	}
+	return opInfo, nil
+}
+
+func (manager *gcfsServiceManager) GetShare(ctx context.Context, obj *Share) (*Share, error) {
+	return &Share{}, nil
+}
+
+func (manager *gcfsServiceManager) ListShares(ctx context.Context, filter *ListFilter) ([]*Share, error) {
+	return nil, nil
+}
+
+func ParseShare(s *Share) (string, string, string, string, error) {
+	if s == nil || s.Parent == nil {
+		return "", "", "", "", fmt.Errorf("Missing parent object for share %s", s.Name)
+	}
+
+	if s.Parent.Project == "" || s.Parent.Location == "" || s.Parent.Name == "" || s.Name == "" {
+		return "", "", "", "", fmt.Errorf("Missing identifier in share")
+	}
+
+	return s.Parent.Project, s.Parent.Location, s.Parent.Name, s.Name, nil
+}
+
+func GetMultishareInstanceHandle(m *MultishareInstance) (string, error) {
+	if m == nil {
+		return "", fmt.Errorf("empty multishare instance")
+	}
+	return fmt.Sprintf("%s/%s/%s", m.Project, m.Location, m.Name), nil
+}
+
+func CompareMultishareInstances(a, b *MultishareInstance) error {
+	if a == nil || b == nil {
+		return fmt.Errorf("empty instance object detected")
+	}
+
+	mismatches := []string{}
+	if a.Project != b.Project {
+		mismatches = append(mismatches, "project")
+	}
+
+	if a.Location != b.Location {
+		mismatches = append(mismatches, "location")
+	}
+
+	if a.Name != b.Name {
+		mismatches = append(mismatches, "name")
+	}
+
+	if strings.ToLower(a.Tier) != strings.ToLower(b.Tier) {
+		mismatches = append(mismatches, "tier")
+	}
+
+	if a.Network.Name != b.Network.Name {
+		mismatches = append(mismatches, "network name")
+	}
+
+	// Filestore API does not include key version info in the Instance object, simple string comparison will work
+	if a.KmsKeyName != b.KmsKeyName {
+		mismatches = append(mismatches, "kms key name")
+	}
+
+	if len(mismatches) > 0 {
+		return fmt.Errorf("instance %v already exists but doesn't match expected: %+v", a.Name, mismatches)
+	}
+	return nil
+}
+
+func CompareShares(a, b *Share) error {
+	if a == nil || b == nil {
+		return fmt.Errorf("empty share object detected")
+	}
+
+	mismatches := []string{}
+	if a.Name != b.Name {
+		mismatches = append(mismatches, "tier")
+	}
+	if util.RoundBytesToGb(a.CapacityBytes) != util.RoundBytesToGb(b.CapacityBytes) {
+		mismatches = append(mismatches, "share size")
+	}
+	if len(mismatches) > 0 {
```

Comment:
do we not need to compare parents as well

Reply:
it's compared in line 752

Reply:
line 752 seems to compare `Name` not `Parent`